pkg/operator/configobservation/etcdendpoints/observe_etcd_endpoints.go

```diff
@@ -0,0 +1,109 @@
+package etcdendpoints
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+	"strings"
+
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	endpointsobserver "github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation/etcd"
+)
+
+const (
+	etcdEndpointNamespace = "openshift-etcd"
+	etcdEndpointName      = "etcd-endpoints"
+)
+
+// BackwardsCompatibleObserveStorageURLs will prefer to observe the new
+// etcd-endpoints configmap, and will fall back to reading the endpoint
+// resource only if the configmap doesn't yet exist.
+func BackwardsCompatibleObserveStorageURLs(genericListers configobserver.Listers, recorder events.Recorder, currentConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	if _, err := listers.ConfigmapLister.ConfigMaps(etcdEndpointNamespace).Get(etcdEndpointName); err == nil {
+		return ObserveStorageURLs(genericListers, recorder, currentConfig)
+	} else if !errors.IsNotFound(err) {
+		return currentConfig, []error{err}
```

Comment:
`	previouslyObservedConfig := map[string]interface{}{}` as done a few lines down.  returning `currentConfig` means a different thing and will result in conflicts with other observers.